posthog-core/src/index.ts

```diff
@@ -202,27 +206,27 @@ export abstract class PostHogCore {
         event: '$identify',
         properties: {
           ...(properties || {}),
-          $anon_distinct_id: this.getDistinctId(), // TODO: Should this be the previous distinct id or the original anon id??
+          $anon_distinct_id: this.getAnonymousId(),
         },
       }),
       $set: properties,
     }
 
-    if (distinctId !== this.getDistinctId()) {
+    if (distinctId !== previousDistinctId) {
+      // We keep the AnonymousId to be used by decide calls and identify to link the previousId
+      this.setPersistedProperty(PostHogPersistedProperty.AnonymousId, previousDistinctId)
```

Comment:
.. oh, this is sortof confusing. This makes the anonymousId be a non-anonymous ID ? (and makes my above comment moot)

Reply:
Yeah its how the posthog-js seems to do it. Just overwrites the anonymousId. That way I guess there is _never_ the chance that it accidentally gets used